x-pack/plugins/saved_objects_tagging/public/management/tag_management_page.tsx

```diff
@@ -0,0 +1,166 @@
+import { Header, TagTable } from './components';
+import { getTagConnectionsUrl } from './utils';
+
+interface TagManagementPageParams {
+  setBreadcrumbs: (crumbs: ChromeBreadcrumb[]) => void;
+  core: CoreStart;
+  tagClient: ITagInternalClient;
+  capabilities: TagsCapabilities;
+}
+
+export const TagManagementPage: FC<TagManagementPageParams> = ({
+  setBreadcrumbs,
+  core,
+  tagClient,
+  capabilities,
+}) => {
+  const { overlays, notifications, application, http } = core;
+  const [loading, setLoading] = useState<boolean>(false);
+  const [allTags, setAllTags] = useState<TagWithRelations[]>([]);
+  const [selectedTags, setSelectedTags] = useState<TagWithRelations[]>([]);
+
+  const createModalOpener = useMemo(() => getCreateModalOpener({ overlays, tagClient }), [
+    overlays,
+    tagClient,
+  ]);
+  const editModalOpener = useMemo(() => getEditModalOpener({ overlays, tagClient }), [
+    overlays,
+    tagClient,
+  ]);
+
+  useEffect(() => {
+    setBreadcrumbs([
+      {
+        text: i18n.translate('xpack.savedObjectsTagging.management.breadcrumb.index', {
+          defaultMessage: 'Tags',
+        }),
+        href: '/',
+      },
+    ]);
+  }, [setBreadcrumbs]);
+
+  const fetchTags = useCallback(async () => {
+    setLoading(true);
+    const { tags } = await tagClient.find({
+      page: 1,
+      perPage: 10000,
+    });
+    setAllTags(tags);
+    setLoading(false);
+  }, [tagClient]);
+
+  useMount(() => {
+    fetchTags();
+  });
+
+  const openCreateModal = useCallback(() => {
+    createModalOpener({
+      onCreate: () => {
+        fetchTags();
+      },
+    });
+  }, [createModalOpener, fetchTags]);
+
+  const openEditModal = useCallback(
+    (tag: TagWithRelations) => {
+      editModalOpener({
+        tagId: tag.id,
+        onUpdate: () => {
+          fetchTags();
+        },
+      });
+    },
+    [editModalOpener, fetchTags]
+  );
+
+  const getTagRelationUrl = useCallback(
+    (tag: TagWithRelations) => {
+      return getTagConnectionsUrl(tag, http.basePath);
+    },
+    [http]
+  );
+
+  const showTagRelations = useCallback(
+    (tag: TagWithRelations) => {
+      application.navigateToUrl(getTagRelationUrl(tag));
+    },
+    [application, getTagRelationUrl]
+  );
+
+  const deleteTagWithConfirm = useCallback(
+    async (tag: TagWithRelations) => {
+      const confirmed = await overlays.openConfirm(
+        i18n.translate('xpack.savedObjectsTagging.modals.confirmDelete.text', {
+          defaultMessage: 'Are you sure you want to delete tag "{name}"?',
+          values: {
+            name: tag.name,
+          },
+        }),
+        {
+          title: i18n.translate('xpack.savedObjectsTagging.modals.confirmDelete.title', {
+            defaultMessage: 'Delete tag',
+          }),
+          confirmButtonText: i18n.translate(
+            'xpack.savedObjectsTagging.modals.confirmDelete.confirmButtonText',
+            {
+              defaultMessage: 'Delete',
```

Comment:
Might be nice to be more explicit here using either `Delete tag` or `Delete {name}`? (Could also then remove the `title` attribute.)

Reply:
Did not have mockups for that

Current design:

<img width="435" alt="Screenshot 2020-10-20 at 18 58 15" src="https://user-images.githubusercontent.com/1532934/96619290-45665d00-1306-11eb-9ca4-81b339faf704.png">

Will let core-ui design what I should put and were, just tell me :)

Reply:
We handle these prompts differently depending relationship and just how destructive the action would be. I think we could follow what has been done with deleting users:

- put the tag name in the title
- use the sentence to clarify this cannot be undone
- leave the buttons as you have them

<img width="438" alt="Screen Shot 2020-10-20 at 1 48 52 PM" src="https://user-images.githubusercontent.com/446285/96631019-6796b580-12db-11eb-9030-6286d398fe21.png">

That said, if we feel like this is more destructive then we could add more detail to better convey the overall impact. For example, we could put the number of related objects and/or spaces ("This will remove the tag from 32 objects across 3 spaces"). My instinct is that this suggestion will be considered a future enhancement.

Reply:
Tags are single namespaced for now, and. getting the info of references across all spaces is not trivial. But I already have the number of connections in the current space as I display it in the table, we can use it in the label if we want. In that case, I would also need the text there is no object assigned to the tag we are deleting.

Reply:
@pgayvallet: FYI, there are indeed design mockups for the delete tag modals (both single and bulk). The single delete tag modal mockup can be found here: https://www.figma.com/file/kh7Gm1iM0zJPj504BeDQC4/Mockups-Version-3-Awaiting-Dev?node-id=101%3A56

![image](https://user-images.githubusercontent.com/3884767/96726512-0ffe5580-1380-11eb-9902-1c19bfbe36b0.png)

Reply:
@MichaelMarcialis sorry then, must have missed it. Will integrate that asap.

Reply:
Done in https://github.com/elastic/kibana/pull/79096/commits/6c4b5f15d718670263fc4cc052c820188161a783, resolving.